homeassistant/components/intellifire/switch.py

```diff
@@ -5,50 +5,69 @@
-from .entity import IntellifireEntity
+from .entity import IntelliFireEntity
 
 
 @dataclass()
-class IntellifireSwitchRequiredKeysMixin:
+class IntelliFireSwitchRequiredKeysMixin:
     """Mixin for required keys."""
 
-    on_fn: Callable[[IntellifireAPILocal], Awaitable]
-    off_fn: Callable[[IntellifireAPILocal], Awaitable]
-    value_fn: Callable[[IntellifirePollData], bool]
+    on_fn: Callable[[IntelliFireDataUpdateCoordinator], Awaitable]
+    off_fn: Callable[[IntelliFireDataUpdateCoordinator], Awaitable]
+    value_fn: Callable[[IntelliFireDataUpdateCoordinator], bool]
 
 
 @dataclass
-class IntellifireSwitchEntityDescription(
-    SwitchEntityDescription, IntellifireSwitchRequiredKeysMixin
+class IntelliFireSwitchEntityDescription(
+    SwitchEntityDescription, IntelliFireSwitchRequiredKeysMixin
 ):
     """Describes a switch entity."""
 
 
-INTELLIFIRE_SWITCHES: tuple[IntellifireSwitchEntityDescription, ...] = (
-    IntellifireSwitchEntityDescription(
+INTELLIFIRE_SWITCHES: tuple[IntelliFireSwitchEntityDescription, ...] = (
+    IntelliFireSwitchEntityDescription(
         key="on_off",
         translation_key="flame",
-        on_fn=lambda control_api: control_api.flame_on(),
-        off_fn=lambda control_api: control_api.flame_off(),
-        value_fn=lambda data: data.is_on,
+        on_fn=lambda coordinator: coordinator.get_control_api().flame_on(),
+        off_fn=lambda coordinator: coordinator.get_control_api().flame_off(),
+        value_fn=lambda coordinator: coordinator.get_read_api().data.is_on,
     ),
-    IntellifireSwitchEntityDescription(
+    IntelliFireSwitchEntityDescription(
         key="pilot",
         translation_key="pilot_light",
         icon="mdi:fire-alert",
-        on_fn=lambda control_api: control_api.pilot_on(),
-        off_fn=lambda control_api: control_api.pilot_off(),
-        value_fn=lambda data: data.pilot_on,
+        on_fn=lambda coordinator: coordinator.get_control_api().pilot_on(),
+        off_fn=lambda coordinator: coordinator.get_control_api().pilot_off(),
+        value_fn=lambda coordinator: coordinator.get_read_api().data.pilot_on,
+    ),
+    IntelliFireSwitchEntityDescription(
+        key="cloud_read",
+        name="Cloud read",
+        on_fn=lambda coordinator: coordinator.set_read_mode(IntelliFireApiMode.CLOUD),
+        off_fn=lambda coordinator: coordinator.set_read_mode(IntelliFireApiMode.LOCAL),
+        value_fn=lambda coordinator: coordinator.get_read_mode()
+        == IntelliFireApiMode.CLOUD,
+    ),
+    IntelliFireSwitchEntityDescription(
+        key="cloud_control",
+        name="Cloud control",
+        on_fn=lambda coordinator: coordinator.set_control_mode(
+            IntelliFireApiMode.CLOUD
+        ),
+        off_fn=lambda coordinator: coordinator.set_control_mode(
+            IntelliFireApiMode.LOCAL
+        ),
+        value_fn=lambda coordinator: coordinator.get_control_mode()
+        == IntelliFireApiMode.CLOUD,
```

Comment:
These are settable by the OptionsFlow, however, I thought it might be nice to have switches. Does this go against any HA paradigms?